### PR TITLE
changed logic for canMissVerification

### PR DESCRIPTION
**Pull request by HMAshiqulIslam**

### CODE REVIEW

File: src/Lachain.Core/Blockchain/Operations/TransactionManager.cs [hunk 1]
```diff
@@ -148,5 +154,64 @@ public ulong CalcNextTxNonce(UInt160 from)
...
+
+        private bool IsDistributeCycleRewardsAndPenaltiesTx(Block block, Transaction tx) 
+        {
+            var indexInCycle = block.Header.Index % StakingContract.CycleDuration;
+            var cycle = block.Header.Index / StakingContract.CycleDuration;
+            if (cycle > 0 && indexInCycle == StakingContract.AttendanceDetectionDuration)
+            {
+                var expectedTx = BuildSystemContractTx(ContractRegisterer.GovernanceContract, GovernanceInterface.MethodDistributeCycleRewardsAndPenalties, 
+                    UInt256Utils.ToUInt256(GovernanceContract.GetCycleByBlockNumber(block.Header.Index - 1)));
+                return expectedTx.Equals(tx);
```
Comment (sgladkov) [inline src/Lachain.Core/Blockchain/Operations/TransactionManager.cs:166]: It looks dangerous to me. Received tx can differ in some not important params like GasPrice or something like this. May be better to check inmportant fields (From,  To,  Invocation,  Value) only?

**Comment by sgladkov** [comment]
Also we need detailed information in the log if we reject system transaction (exact reason and tx itself)

File: src/Lachain.Core/Blockchain/Operations/TransactionManager.cs (hunk 1 above)
Comment (HMAshiqulIslam) [reply src/Lachain.Core/Blockchain/Operations/TransactionManager.cs:166]: Nice Catch. I totally missed this. I will push a fix.

**Comment by HMAshiqulIslam** [comment]
> Also we need detailed information in the log if we reject system transaction (exact reason and tx itself)

**Comment by HMAshiqulIslam** [comment]
> Also we need detailed information in the log if we reject system transaction (exact reason and tx itself)
I agree. I will write a fix. Thanks.

**Review by sgladkov — APPROVED** [review]

**Pull request merged**